cni-plugin/pkg/ipamplugin/ipam_plugin.go

```diff
@@ -252,6 +255,26 @@ func cmdAdd(args *skel.CmdArgs) error {
 			logrus.Info("Running in single-HNS-network mode, limiting number of IPAM blocks to 1.")
 			maxBlocks = 1
 		}
+		// Get namespace information for namespaceSelector support
+		namespace := epIDs.Namespace
+		var namespaceObj *corev1.Namespace
+
+		// Only attempt to fetch namespace if we have Kubernetes configuration and a valid namespace
+		if (conf.Kubernetes.Kubeconfig != "" || conf.Policy.PolicyType == "k8s") && namespace != "" {
+			logger.Debugf("Getting namespace for: %s", namespace)
+
+			ns, err := getNamespace(conf, namespace, logger)
```

Comment:
```suggestion
			namespaceObj, err = getNamespace(conf, namespace, logger)
```

You can assign directly to the `namespaceObj` variable and it will avoid the need for the `else` block below.